### DISCUSSION TITLE
Error Tensorflow pada fit() untuk kasus binary crossentropy

### DISCUSSION THREAD

**Discussion by MelissaRusli** [teamalgoritma/community#813] [Q&A] [answered]
Saya mengalami kendala error pada tahap pembuatan fitting model untuk Neural Network seperti dibawah ini:

<img width="908" alt="Screen Shot 2023-11-30 at 5 09 18 AM" src="https://github.com/teamalgoritma/community/assets/141122159/6e6ecfc3-19e0-4915-81ee-bd633faee799">

NN model yang saya coba buat adalah klasifikasi biner (2 output value) dengan step2 sebelumnya seperti berikut:

<img width="1223" alt="Screen Shot 2023-11-30 at 5 10 09 AM" src="https://github.com/teamalgoritma/community/assets/141122159/675bb54b-60ad-4316-a4fd-d6916e3c7607">

Mohon arahan nya atas error yang terjadi  🙏🏻

Sebagai info juga, komputer yang saya pakai adalah M1 , tetapi proses sebelum fitting model ini tidak mengalami kendala ketika di running, hanya setiap kali restart R project ini, saya perlu mengulangi proses tahapan reticulate dan library(tensorflow) pada console.

Terima Kasih.

**Comment by kinantytasyao** [comment]
Dear @MelissaRusli 

Untuk menelusuri masalah yang terjadi, apakah boleh discreenshot tahapan data preprocessing sampai splitting data nya Bu?

**Comment by MelissaRusli** [comment]
Berikut tahapan process yang saya lakukan :

<img width="1138" alt="Screen Shot 2023-11-30 at 11 37 13 AM" src="https://github.com/teamalgoritma/community/assets/141122159/0a0170fd-99d5-4d5a-8088-83dd87d57faa">

<img width="1175" alt="Screen Shot 2023-11-30 at 11 37 47 AM" src="https://github.com/teamalgoritma/community/assets/141122159/3ff4dfbe-cbdd-43fd-b224-c7ec40e0ee0d">

<img width="1177" alt="Screen Shot 2023-11-30 at 11 38 12 AM" src="https://github.com/teamalgoritma/community/assets/141122159/205d7e67-d65e-40b8-86e4-f46fa5ffb76c">

<img width="1190" alt="Screen Shot 2023-11-30 at 11 38 30 AM" src="https://github.com/teamalgoritma/community/assets/141122159/fac5c110-0efb-47db-b7b1-e981925ea88f">

<img width="1181" alt="Screen Shot 2023-11-30 at 11 38 46 AM" src="https://github.com/teamalgoritma/community/assets/141122159/d20a0dcc-60d6-4baa-a16a-db684f00ec81">

**Reply by kinantytasyao** [reply, marked as answer]
Error di atas biasa terjadi ketika terdapat ketidaksesuaian input model. Meninjau data preprocessing yang Ibu lampirkan, sepertinya ada kekeliruan dalam mentreatment variabel prediktor kategorik, seharusnya variabel kategorik yang ada diencode dulu menggunakan one hot encoding sebelum dimasukkan ke dalam model karena model NN tidak dapat melakukan perhitungan matematis jika inputnya bukan angka.

Demikian jawaban yang dapat kami sampaikan. Semoga membantu :)

Jika ternyata jawaban ini menyelesaikan permasalahan Anda, silakan mark as anwer :)

**Reply by MelissaRusli** [reply]
Terima Kasih atas arahan nya 😊 🙏🏻

**Reply by kinantytasyao** [reply]
Sama sama Bu Melissa :D
